.github/actions/validate-repo-version/action.yml

```diff
@@ -23,27 +23,27 @@ runs:
         version=$(jq --compact-output --raw-output '."${{ inputs.repo-to-check }}"' ./config/versions.json)
 
         if [[ "${version}" == "null" ]]; then
+          echo "::error::There is no `${{ inputs.repo-to-check }}` in `./config/versions.json`"
           exit 1
-        else
-          echo "version=${version}" >> $GITHUB_OUTPUT
         fi
 
+        echo "version=${version}" >> $GITHUB_OUTPUT
+        echo "spack-branch-version=releases/v${version}" >> $GITHUB_OUTPUT
+
     # Verify that the repository exists at the given ref
     - name: Version Check
       id: check
       uses: actions/checkout@v4
       with:
         repository: access-nri/${{ inputs.repo-to-check }}
-        ref: ${{ steps.versions.outputs.packages }}
+        # In the case where we are checking spack, we need to add the
+        # 'releases/v' to the ref name, obtained in the previous step
+        ref: ${{ inputs.repo-to-check == 'spack' && steps.jq.outputs.spack-branch-version || steps.jq.outputs.version }}
         path: repo
 
-    - name: Failure Notifier
-      if: failure()
+    - name: Version Check Failure Notifier
+      if: failure() && steps.check.outcome == 'failure'
```

Comment:
Is only `steps.check.outcome == 'failure'` needed here?

Reply:
Yeah, because we fail earlier in the workflow for the other fail conditions